File: circle_work_backend/src/logic/objects.ts
import internal from "stream";

export interface Goal {
    id: number
    owner_id: number
    owner_name: string
    is_private: boolean
    goal_name: string
    goal_body: string
    approval_count: number
    categoryId: number
    due_date: string
}

export interface Circle {
    id: number
    name: string
    color: number
    team_count: number
    monthly_donation: number
    total_donation: number
    tasks_started: number
    tasks_completely: number
}

export interface GoalCategory{
    id: number
    name: string
    description: string
    color: number
}

export interface Charity{
    id: number
    name: string
    image: string
    link: string
    raised_monthly: number
    raised_total: number
}

export interface Payment{
    id: number
    user_id: number
    amount: number
}

export interface CategoryResources{
    id: number
    category: number
    title: string
    link: string
    type: number
}

export interface DonationCounts {
    total: number
    month: number
}

File: CircleWork/src/main/java/com/circlework/Objects.java
package com.circlework;

import java.sql.Timestamp;

public class Objects {

    //
    // IF YOU CHANGE ANY OF THESE, UPDATE THE objects.ts FILE TOO!!!!!!!!
    //

    public record Empty() {}

    public record User(
        int id,
        String username,
        String password,
        int balance,
        int circle_id) {}

//1078

    public record Goal(
            int id,
            int owner_id,
            String owner_name,
            boolean is_private,
            int categoryId,
            String goal_name,
            String goal_body,
            Timestamp due_date,
            int approval_count
            ) {}

    public record Circle(
        int id,
        String name,
        int color,
        int team_count,
        int monthly_donation,
        int total_donation,
        int tasks_started,
        int tasks_completed) {}

    public record GoalCategory(
        int id,
        String name,
        String description,
        int color
    ) {}

    public record Charity(
        int id,
        String name,
        String image,
        String link,
        int raised_monthly,
        int raised_total
    ) {}

    public record Payment(
        int id,
        int user_id,
        int amount
    ) {}

    public record CategoryResources(
        int id,
        int category,
        String title,
        String link,
        int type
    ) {}
}
